ColorPalette: white palette color merging with white background

lukekowalski opened the pull request:
ColorPalette block allows to provide custom colors but in case we have white (#fff) color among them it may be confusing for the users as it will merge with the background.

This PR solves it by adding a thin border around the white colour so it stands out (border color matches the border from '.components-panel__color-area' class)


## How Has This Been Tested?
Insert ColorPalette component in some part of the editor i.e: existing 'paragraph' block for test purposes and verify that white colour stands out.
Sample: 
```javascript
<PanelColor title={ __( 'Text Color' ) } colorValue={ textColor } initialOpen={ false }>
	<ColorPalette
		colors={ [ '#fff', '#000', '#ccc' ] }
		value={ textColor }
		onChange={ ( colorValue ) => setAttributes( { textColor: colorValue } ) }
	/>
</PanelColor>
```

## Screenshots (jpeg or gifs if applicable):

Before fix:
![image](https://user-images.githubusercontent.com/36142579/37379268-b6680bf4-272a-11e8-8807-8f6443fbcfff.png)

After fix:
![image](https://user-images.githubusercontent.com/36142579/37379273-bb126866-272a-11e8-92be-39c44e9e8632.png)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.

Commit:
White colour on white background

Fixing white color from ColorPalette block merging with the background

bordoni commented:
Why not add  transparent border for all with 20% opacity from black? it removes the conditional but and applies to colors that are very close to white too. like `#efefef`

Commit:
Adding border to all colors

Adding black border with 20% opacity to all colors to cover multiple scenarios (all colors that are close to white + multiple ways of declaring white in the ColorPalette)

wpjsio commented:
That's a very good point. I guess I've seen `border:none` there and I just focused on not altering the initial designs too much. Your approach is much better though as not only doesn't require conditional but also it covers multiple scenarios and ways to declare white color in the ColorPalette (ie: #fff, #ffffff, white ). Applied this small change. 

Preview available below:
![image](https://user-images.githubusercontent.com/36142579/37419064-922133a2-27ab-11e8-940d-cde8bf3980e8.png)

paulwilde commented:
Using a pseudo class to add an inset box-shadow would allow for the border to match the same colour each option, as otherwise the grey kinda looks bad against darker colours.

```
.blocks-color-palette__item:after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    border-radius: 50%;
    box-shadow: inset 0 0 0 1px rgba(0, 0, 0, .2);
}
```

<img width="280" alt="screen shot 2018-03-15 at 09 13 28" src="https://user-images.githubusercontent.com/704594/37453944-2aef9b5c-2831-11e8-91ee-3e59d71325ce.png">

wpjsio commented:
That's super clever. Thanks for digging into it!

Commit:
Replacing border with box-shadow

inset box-shadow allows for the border to match the same colour as each option

karmatosed commented:
@paulwilde assuming this works on all colors this is the preferred way for me: https://github.com/WordPress/gutenberg/pull/5598#issuecomment-373310465

Thanks for digging into this everyone.
